src/main/java/seedu/address/model/EntityList/ParticipantList.java

```diff
@@ -20,56 +20,54 @@ public ParticipantList() {
        this.participants = new ArrayList<>();
     }
 
-    /**
-     * Gets participant by id.
-     *
-     * @param id
-     * @return
-     */
-    @Override
-    public Participant get(Id id) {
-        return new Participant(new Name("name"), new Email("email"), new Phone("999"), this.generateID());
-    }
-
-    /**
-     * Updates participant by id.
-     *
-     * @param participant
-     * @throws Exception if error while updating.
-     */
-    @Override
-    public void update(Entity participant) throws Exception {
-        // TODO: Find by ID then replace.
-        participants.add((Participant) participant);
-    }
-
-    /**
-     * Adds participant to the list.
-     *
-     * @param participant
-     * @throws Exception if there was an error while adding.
-     */
-    @Override
-    public void add(Entity participant) throws Exception {
-        // TODO: Find by ID then replace.
-        participants.add((Participant) participant);
-    }
-
-    /**
-     * Deletes participant by ID.
-     *
-     * @param id
-     * @throws Exception if error while deleting.
-     */
-    @Override
-    public void delete(Id id) throws Exception {
-        for (Participant p: this.participants) {
-            if (p.getId() == id) {
-                this.participants.remove(p);
-                return;
-            }
-        }
-    }
+//    /**
+//     * Gets participant by id.
+//     *
+//     * @param id
+//     * @return
+//     */
+//    @Override
+//    public Participant get(Id id) {
+//        return new Participant(new Name("name"), new Email("email"), new Phone("999"), this.generateID());
+//    }
+//
+//    /**
+//     * Updates participant by id.
+//     *
+//     * @param participant
+//     * @throws Exception if error while updating.
+//     */
+//    @Override
+//    public void update(Entity participant) throws Exception {
+//        participants.add((Participant) participant);
+//    }
+//
+//    /**
+//     * Adds participant to the list.
+//     *
+//     * @param participant
+//     * @throws Exception if there was an error while adding.
+//     */
+//    @Override
+//    public void add(Entity participant) throws Exception {
+//        participants.add((Participant) participant);
+//    }
+//
+//    /**
+//     * Deletes participant by ID.
+//     *
+//     * @param id
+//     * @throws Exception if error while deleting.
+//     */
+//    @Override
+//    public void delete(Id id) throws Exception {
+//        for (Participant p: this.participants) {
+//            if (p.getId() == id) {
+//                this.participants.remove(p);
+//                return;
+//            }
+//        }
+//    }
 
     /**
      * List the participants.
```

Comment:
Since we want to make this read only, I think you should return a clone of the list or a stream of the list.

Reply:
The individual elements still need to be deep copied tho, but yea returning a stream sounds good, I'll do that. But yea for this method just avoid mutating the list in the code.

Reply:
hmm yea I'll return a stream, but the individual references are still not deep copied, so its kinda read only in name. Hence, must avoid mutating the list directly and use the `Model` API instead

Reply:
I needed it for the `list` command, so it should be fine